core/src/com/mygdx/game/models/TerrainClass.java

```diff
@@ -0,0 +1,71 @@
+package com.mygdx.game.models;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.files.FileHandle;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+
+/**
+ * A class that represents a tile in the game.
+ */
+public class TerrainClass implements Cloneable, Terrain {
+    private String type;
+    private int defense;
+    private TerrainState terrainState;
+
+    private TerrainClass(String type, int defense, TerrainState terrainState) {
+        this.type = type;
+        this.defense = defense;
+        this.terrainState = terrainState;
+    }
+
+    // Copy constructor
+    public TerrainClass(TerrainClass original) {
+        type = original.type; // Immutable
+        defense = original.defense; // Immutable
+        terrainState = original.terrainState; // Not sure if this is the proper way to copy enum
+    }
+
+    /**
+     * Returns the Terrain from the configuration file.
+     *
+     * @param filename the filename
+     * @return the Terrain from the configuration file
+     */
+    public static TerrainClass getTerrainFromConfig(String filename) {
```

Comment:
Can you add this method to the Configuration Factory. In the configuration factory, we should be saving Terrain objects based on the configuration name in a Hashmap. Then when someone wants to make a new Terrain, if we have already saved that Terrarina type, then we just get it from the map. Else we use this method to parse the Configuration.

Reply:
Something like this?
public TerrainClass makeTerrainFromConfig(String config) {
    if (terrainHashMap.containsKey(config)) return terrainHashMap.get(config);
    else return TerrainClass.getTerrainFromConfig(config);
}